SchedulerJobs/SchedulerJobs.Services/AzureStorageServiceBase.cs

```diff
@@ -0,0 +1,216 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Storage;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Sas;
+using SchedulerJobs.Common.Exceptions;
+using SchedulerJobs.Services.Configuration;
+
+namespace SchedulerJobs.Services
+{
+    public class AzureStorageServiceBase
+    {
+        private readonly BlobServiceClient _serviceClient;
+
+        private readonly IBlobStorageConfiguration _blobStorageConfiguration;
+
+        private readonly bool _useUserDelegation;
+
+        private readonly IBlobClientExtension _blobClientExtension;
+
+
+        protected AzureStorageServiceBase(BlobServiceClient serviceClient,
+            IBlobStorageConfiguration blobStorageConfiguration, IBlobClientExtension blobClientExtension,
+            bool useUserDelegation)
+        {
+            _serviceClient = serviceClient;
+            _blobStorageConfiguration = blobStorageConfiguration;
+            _useUserDelegation = useUserDelegation;
+            _blobClientExtension = blobClientExtension;
+        }
+
+        /// <summary>
+        /// Upload file to the azure storage container.
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <param name="file"></param>
+        public async Task UploadFile(string fileName, byte[] file)
+        {
+            var containerClient = _serviceClient.GetBlobContainerClient(_blobStorageConfiguration.StorageContainerName);
+
+            var blobClient = containerClient.GetBlobClient(fileName);
+
+            await using MemoryStream ms = new MemoryStream(file);
+            await blobClient.UploadAsync(ms);
+        }
+
+        /// <summary>
+        /// Delete all blobs from the azure storage container.
+        /// </summary>
+        public async Task ClearBlobs()
+        {
+            var blobNames = await GetAllBlobNamesByFilePathPrefix(".json");
```

Comment:
Dosen't look like we're using this anywehre 'blobNames'

Reply:
Agree - think this line can be deleted as that method is clearing all blobs

Reply:
you are right. I have removed the line now